tool_collections/samtools/bam_to_cram/samtools_bam_to_cram.xml

```diff
@@ -0,0 +1,125 @@
+    <![CDATA[
+        #if str( $input_alignment.metadata.bam_index ) != "None":
+            ln
+                -f
+                -s
+                "${input_alignment.metadata.bam_index}"
+                "${input_alignment}.bai"
+                &&
+        #end if
+
+        samtools view
+            #if $parameter_regions.target_region == "regions_bed_file"
+                -L $parameter_regions.regions_bed_file
+            #end if
+            
+            -@\${GALAXY_SLOTS:-2}
+            -C
+            -T ${reference_source.input_reference}
+            -o $output_alignment
+            $input_alignment
+            
+            #if $parameter_regions.target_region == "region"
+                "${parameter_regions.region_string}"
+            #end if
+    ]]>
+    </command>
+
+    <inputs>
+        <param name="input_alignment" type="data" format="bam,sam" label="BAM (or SAM) alignment file"/>
+        <conditional name="reference_source">
+            <param name="reference_source_selector" type="select" label="Load reference genome from">
+                <option value="cached">Local cache</option>
+                <option value="history">History</option>
+            </param>
+            <when value="cached">
+                <param name="input_reference" type="select" format="fasta" label="Genome reference FASTA file">
+                    <options from_data_table="all_fasta">
+                        <filter type="data_meta" ref="input_alignment" key="dbkey" column="1" />
+                    </options>
+                    <validator type="no_options" message="A built-in reference genome is not available for the build associated with the selected input file"/>
+                </param>
+            </when>
+            <when value="history">
+                <param name="input_reference" type="data" format="fasta" label="Genome reference FASTA file"/>
+            </when>
+        </conditional>
+        <conditional name="parameter_regions">
+            <param name="target_region" type="select" label="Choose conversion within specific genomic region(s)">
+                <option value="entire_input_file">Entire BAM alignment file</option>
+                <option value="region">Specific region</option>
+                <option value="regions_bed_file">List of specific regions (BED file)</option>
+            </param>
+            <when value="entire_input_file" />
+            <when value="region">
+                <param name="region_string" type="text" label="Samtools: region in which pileup is generated" help="e.g. chrX or chr:start-end" />
+            </when>
+            <when value="regions_bed_file">
+                <param name="regions_bed_file" type="data" format="bed" label="only include reads overlapping this BED FILE" help="-L"/>
+            </when>
+        </conditional>
+    </inputs>
+
+    <outputs>
+        <data format="cram" name="output_alignment" label="$tool.name on ${on_string}.cram"></data>
```

Comment:
I have tested how it currently works without having changed the code.

Test1:
* input_alignment (input bam/cram): hg19
* input_reference (fasta): unset
* output_alignment: hg19

Test2:
* input_alignment (input bam/cram): unset
* input_reference (fasta): unset
* output_alignment: unset

Test3:
* input_alignment (input bam/cram): unset
* input_reference (fasta): hg19
* output_alignment: hg19

Test4:
* input_alignment (input bam/cram): hg19
* input_reference (fasta): hg18
* output_alignment: hg19

I suppose that the first 3 scenarios are as they should be. The last one can be arguable, although picking the dbkey of the input alignment has my preference since the goal is to convert the alignment (also using its metadata). @nsoranzo If you prefer to change it or make the dbkey behaviour static, please let me know. Thanks for looking into this!